ioc.types.ts

```diff
@@ -0,0 +1,18 @@
+import MessageHandlerFactory from "./source/MessageHandlerFactory";
+
+let TYPES = {
```

Comment:
Instead of resolving to concrete types we can generally use tyese 'enums' to lookup bindings to resolve when doing a container.get.